File: sample-conference-java/app/src/main/java/com/quickblox/sample/videochat/conference/java/utils/ToastUtils.java
package com.quickblox.sample.videochat.conference.java.utils;

import android.content.Context;
import android.widget.Toast;

import androidx.annotation.StringRes;

public class ToastUtils {

    private ToastUtils() {
    }

    public static void shortToast(Context context, String message) {
        show(context, message, Toast.LENGTH_LONG);
    }

    public static void shortToast(Context context, @StringRes int resource) {
        show(context, context.getString(resource), Toast.LENGTH_SHORT);
    }

    public static void longToast(Context context, String message) {
        show(context, message, Toast.LENGTH_LONG);
    }

    public static void longToast(Context context, @StringRes int resource) {
        show(context, context.getString(resource), Toast.LENGTH_LONG);
    }

    private static void show(Context context, String message, int length) {
        Toast.makeText(context, message, length).show();
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/utils/ToastUtils.kt
package com.quickblox.sample.chat.kotlin.utils

import android.widget.Toast
import androidx.annotation.IntDef
import androidx.annotation.StringRes
import com.quickblox.sample.chat.kotlin.App

@IntDef(Toast.LENGTH_LONG, Toast.LENGTH_SHORT)
private annotation class ToastLength

fun shortToast(@StringRes text: Int) {
    shortToast(App.instance.getString(text))
}

fun shortToast(text: String?) {
    show(text, Toast.LENGTH_SHORT)
}

fun longToast(@StringRes text: Int) {
    longToast(App.instance.getString(text))
}

fun longToast(text: String) {
    show(text, Toast.LENGTH_LONG)
}

private fun show(text: String?, @ToastLength length: Int) {
    text?.let {
        makeToast(it, length).show()
    }
}

private fun makeToast(text: String, @ToastLength length: Int): Toast {
    return Toast.makeText(App.instance, text, length)
}